pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -146,8 +153,38 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 				return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token for serviceaccount %q to pod running with different serviceaccount name.", name))
 			}
 			uid = pod.UID
+			if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
+				if nodeName := pod.Spec.NodeName; nodeName != "" {
+					newCtx := newContext(ctx, "nodes", nodeName, "", api.SchemeGroupVersion.WithKind("Node"))
+					// set ResourceVersion=0 to allow this to be read/served from the apiservers watch cache
+					nodeObj, err := r.nodes.Get(newCtx, nodeName, &metav1.GetOptions{ResourceVersion: "0"})
```

Comment:
Looking at

https://github.com/kubernetes/kubernetes/blob/f8a4e343a106a73145464e8de8a919d13b59d25a/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L668-L718

this will fallback to live lookups on some branches, should we just fallback to live lookup on all errors (while continuing to ignore `NotFound` errors)?

Reply:
fallback is fine with me